tests/run-make/fmt-write-bloat/main.rs

```diff
@@ -5,7 +5,9 @@
 use core::fmt;
 use core::fmt::Write;
 
-#[link(name = "c")]
+#[cfg_attr(not(target_env = "msvc"), link(name = "c"))]
+#[cfg_attr(all(target_env = "msvc", target_feature = "crt-static"), link(name = "libcmt"))]
+#[cfg_attr(all(target_env = "msvc", not(target_feature = "crt-static")), link(name = "msvcrt"))]
```

Comment:
Remark: this is not obvious, at all lol

Reply:
Aye. I do think this logic should ideally either be in run-make-support or better yet compiletest could have something like a `NO_STD_EXTRA_ARGS` variable that any test can use. Or at least I think that's better than individual tests needing to figure it out. But I'm not entirely sure how best to approach that.

Reply:
Good point. I opened #128821 so this doesn't get lost in a review comment once the PR is merged.